tests/unit/test_snapshot.py

```diff
@@ -1314,6 +1343,19 @@ def test_partition_query_other_error(self):
             attributes=dict(BASE_ATTRIBUTES, **{"db.statement": SQL_QUERY}),
         )
 
+    def test_partition_query_should_throw_error_for_recycled_session(self):
```

Comment:
same comment as partioned read. we would never have this situation practically right?